src/client/mesh.cpp

```diff
@@ -328,6 +328,22 @@ void recalculateBoundingBox(scene::IMesh *src_mesh)
 	src_mesh->setBoundingBox(bbox);
 }
 
+bool checkMeshNormals(scene::IMesh *mesh)
+{
+	u32 buffersCount = mesh->getMeshBufferCount();
+
+	for (u32 i = 0; i < buffersCount; i++) {
+		scene::IMeshBuffer *buffer = mesh->getMeshBuffer(i);
+		core::vector3df firstNormal = buffer->getNormal(0);
```

Comment:
I intentionally check only first normal, assuming that if buffer has it valid, then most likely all other ones are fine too. We can check all of the normals to have length, but it seems like an overkill hurting the performance and covering only really weird broken models.

Reply:
It would be good to write this down in a (short) source code comment.

Reply:
Fair enough